src/hotspot/share/code/codeCache.cpp

```diff
@@ -324,34 +307,44 @@ void CodeCache::initialize_heaps() {
     }
   }
 
+  // last adjustment: leftovers from page alignment go to non_nmethod segment
+  non_nmethod.size += non_profiled.size & alignment_mask(min_size);
+  non_nmethod.size += profiled.size & alignment_mask(min_size);
+
   // Note: if large page support is enabled, min_size is at least the large
   // page size. This ensures that the code cache is covered by large pages.
-  non_nmethod_size = align_up(non_nmethod_size, min_size);
-  profiled_size    = align_down(profiled_size, min_size);
-  non_profiled_size = align_down(non_profiled_size, min_size);
-
-  // Reserve one continuous chunk of memory for CodeHeaps and split it into
-  // parts for the individual heaps. The memory layout looks like this:
-  // ---------- high -----------
-  //    Non-profiled nmethods
-  //         Non-nmethods
-  //      Profiled nmethods
-  // ---------- low ------------
+  non_nmethod.size = align_up(non_nmethod.size, min_size);
+  non_profiled.size = align_down(non_profiled.size, min_size);
+  profiled.size = align_down(profiled.size, min_size);
+
+  FLAG_SET_ERGO(NonNMethodCodeHeapSize, non_nmethod.size);
+  FLAG_SET_ERGO(ProfiledCodeHeapSize, profiled.size);
+  FLAG_SET_ERGO(NonProfiledCodeHeapSize, non_profiled.size);
+  FLAG_SET_ERGO(ReservedCodeCacheSize, cache_size);
```

Comment:
A potential problem here: code heap sizes are `min_size` aligned but `cache_size` might be not.

Reply:
I added an alignment. thanks